### PR TITLE
feat(vectordb): add a vectordb configuration for ai application

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?
type-feat

#### What this PR does / why we need it:
add a vectordb configuration for ai application
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by yizhouxw — COMMENTED** [review]
lgtm

**Review by yizhouxw — COMMENTED** [review]
lgtm

**Review by yizhouxw — COMMENTED** [review]
wrong target branch ?

File: server/odc-service/src/main/java/com/oceanbase/odc/config/VectorDBConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                new TaskThreadFactory("vectordb-bootstrap-executor"), new ThreadPoolExecutor.CallerRunsPolicy());
+    }
+
+    @PreDestroy
+    public void destroy() {
+        ExecutorUtils.gracefulShutdown(vectorDbBootstrapExecutor, "vectordb-bootstrap-executor", 5);
+    }
+
+    @Bean(name = "vectordbDataSource")
+    public DataSource vectordbDataSource() {
```
[the remark this answers is not in the document]
Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/config/VectorDBConfiguration.java:81]: done

**Comment by yhilmare** [comment]
> wrong target branch ?

yes, I have changed it

**Review by yizhouxw — APPROVED** [review]
lgtm

**Review by yizhouxw — APPROVED** [review]

**Pull request merged**